test/mlabns/mlabns.cpp

```diff
@@ -35,12 +38,198 @@ TEST_CASE("Query can pass the settings to the dns level") {
     std::string tool = "neubot";
 
     loop_with_initial_event([=]() {
-        mlabns::query(
-            tool, [](Error error, mlabns::Reply reply) {
-                REQUIRE(error);
+        mlabns::query(tool,
+                      [](Error error, mlabns::Reply) {
+                          REQUIRE(error);
+                          break_loop();
+                      },
+                      settings);
+    });
+}
+
+TEST_CASE("Make sure that an error is passed to callback with invalid query") {
+    Settings settings;
+    settings["mlabns/address_family"] = "ip4"; // Invalid
+    settings["mlabns/metro"] = "trn";
+    settings["mlabns/policy"] = "random";
+    std::string tool = "neubot";
+
+    loop_with_initial_event([=]() {
+        mlabns::query(tool,
+                      [](Error error, mlabns::Reply) {
+                          REQUIRE(error);
+                          break_loop();
+                      },
+                      settings);
+    });
+}
+
+TEST_CASE(
+    "Make sure that an error is passed to callback with invalid query 1") {
+    Settings settings;
+    settings["mlabns/address_family"] = "ipv4";
+    settings["mlabns/metro"] = "trno"; // Invalid
+    settings["mlabns/policy"] = "random";
+    std::string tool = "neubot";
+
+    loop_with_initial_event([=]() {
+        mlabns::query(tool,
+                      [](Error error, mlabns::Reply) {
+                          REQUIRE(error);
+                          break_loop();
+                      },
+                      settings);
+    });
+}
+
+TEST_CASE(
+    "Make sure that an error is passed to callback with invalid query 2") {
+    Settings settings;
+    settings["mlabns/address_family"] = "ipv4";
+    settings["mlabns/metro"] = "trn";
+    settings["mlabns/policy"] = "antani"; // Invalid
+    std::string tool = "neubot";
+
+    loop_with_initial_event([=]() {
+        mlabns::query(tool,
+                      [](Error error, mlabns::Reply) {
+                          REQUIRE(error);
+                          break_loop();
+                      },
+                      settings);
+    });
+}
+
+TEST_CASE(
+    "Make sure that an error is passed to callback with invalid query 3") {
+    Settings settings;
+    settings["mlabns/address_family"] = "ipv4";
+    settings["mlabns/metro"] = "trn";
+    settings["mlabns/policy"] = "random";
+    std::string tool = "antani"; // Invalid
+
+    loop_with_initial_event([=]() {
+        mlabns::query(tool,
+                      [](Error error, mlabns::Reply) {
+                          REQUIRE(error);
+                          break_loop();
+                      },
+                      settings);
+    });
+}
+
+static void get_debug_error(std::string, http::RequestCallback cb,
+                            http::Headers, std::string, Settings, Var<Logger>,
+                            Var<Reactor>) {
+    cb(GenericError(), http::Response());
```

Comment:
My comment is that, when you need to mock the failure of an API for testing purposes, you should not emit `GenericError` because that kind of error is also emitted by MK in many other cases, rather you should emit `MockedError` which shall only be used for indicating that the failure of an API is currently being mocked.